flow_control/tests/test_flowcontrol.py

```diff
@@ -98,29 +99,38 @@ def test_get_location_string(self):
         It should return the problem location string given its Id
         """
 
+        course_strings = [
+            'course',
+            'course-v1',
+            'Course',
+            'course+course',
+        ]
+
         # prepare
         resource = 'problem'
         locator = 'hbdf3883be0935'
         course_prefix = 'course'
-        resource = 'problem'
-        self.block.course_id = MagicMock()
-        self.block.course_id.BLOCK_PREFIX = 'dsgdseg'
-        self.block.course_id.BLOCK_TYPE_PREFIX = 'dfsgesh'
-        course_url = self.block.course_id.to_deprecated_string()
-        course_url = course_url.split(course_prefix)[-1]
-        # execute code
-        result_string = self.block.get_location_string(locator)
 
-        # asserts
-        testing_string = '{prefix}{couse_str}+{type}@{type_id}+{prefix}@{locator}'.format(
-            prefix=self.block.course_id.BLOCK_PREFIX,
-            couse_str=course_url,
-            type=self.block.course_id.BLOCK_TYPE_PREFIX,
-            type_id=resource,
-            locator=locator)
-
-        self.assertEqual(testing_string, result_string)
-        self.block.course_id.to_deprecated_string.assert_called_with()
+        # We want to pass arbitrary times for random.choice.
+        for index in range(5):  # pylint: disable=unused-variable
+            self.block.course_id = MagicMock()
+            self.block.course_id.BLOCK_PREFIX = random.choice(course_strings)
```

Comment:
Why are you making this random? is it not enough to go through all the course_strings one by one?

Reply:
This random is just for get different strings for BLOCK_PREFIX and BLOCK_TYPE_PREFIX. 
Could be made it in other way? @felipemontoya